common/src/main/java/com/wynntils/wc/utils/lootrun/LootrunUtils.java

```diff
@@ -185,29 +185,29 @@ private static void renderChests(
     private static void renderPoints(
             PoseStack poseStack,
             MultiBufferSource.BufferSource source,
-            Long2ObjectMap<List<List<ColoredPoint>>> points,
+            Long2ObjectMap<List<ColoredPath>> points,
             long chunkLong) {
-        List<List<ColoredPoint>> locations = points.get(chunkLong);
+        List<ColoredPath> locations = points.get(chunkLong);
 
         Level level = McUtils.mc().level;
         if (level == null) return;
 
-        for (List<ColoredPoint> locationsInRoute : locations) {
+        for (ColoredPath locationsInRoute : locations) {
             VertexConsumer consumer = source.getBuffer(LootrunRenderType.LOOTRUN_LINE);
             Matrix4f lastMatrix = poseStack.last().pose();
             boolean sourceBatchEnded = false;
 
-            List<ColoredPoint> toRender = new ArrayList<>();
+            ColoredPath toRender = new ColoredPath(new ArrayList<>());
 
             boolean pauseDraw = false;
             BlockPos lastBlockPos = null;
 
-            for (ColoredPoint point : locationsInRoute) {
+            for (ColoredPoint point : locationsInRoute.points()) {
                 BlockPos blockPos = new BlockPos(point.vec3());
 
                 if (blockPos.equals(lastBlockPos)) { // Do not recalculate block validness
-                    if (!toRender.isEmpty()) {
-                        toRender.add(point);
+                    if (!toRender.points().isEmpty()) {
+                        toRender.points().add(point);
                     }
```

Comment:
What is `toRender`, and why does the above check and lines of code exist? Maybe @coehlrich knows.